doc/modules/bayesian_gaussian_mixture_derivation.rst

```diff
@@ -0,0 +1,628 @@
+.. _bayesian_gaussian_mixture_derivation:
+
+.. currentmodule:: sklearn.mixture
+
+Bayesian Gaussian Mixture Models
+================================
+**If you're not interested in learning how to derive similar algorithms yourself
+and/or you're not interested in changing/debugging the implementation in the
+scikit this document is not for you.**
+
+This document document presents the equation behind the implementation of
+the :class:`BayesianGaussianMixture` class.
+The API is identical to that of the :class:`GaussianMixture` class.
+
+The inference algorithm is the one from the following book:
+
+    * `Pattern recognition and machine learning
+      <http://www.springer.com/kr/book/9780387310732>`_,
+      Bishop, Christopher M. Springer, Vol. 4 No. 4, 2006.
+
+While the book presents the parts of the variationnl inference algorithm, it
+does not go into detail in the mixture modeling part, which can be just as
+complex, or even more. For this reason we present here a full derivation of the
+inference algorithm and all the update and lower-bound equations.
+
+This implementation is expected to scale at least as well as EM for
+the Gaussian mixture.
+
+Update rules for Variational Bayesian inference
+===============================================
+
+Here the full mathematical derivation of the Variational Bayes update
+rules for Gaussian Mixture Models is given.
+
+Notation
+--------
+During this document, we will use the following statistics :
+
+.. math::
+   :nowrap:
+
+   \begin{eqnarray*}
+      N_k & = & \sum_{n=1}^N {r_{nk}} \\
```

Comment:
I find it slightly unusual and therefore confusing but I am fine with keeping it if others agree.